.github/workflows/k8s_integration_test.yml

```diff
@@ -0,0 +1,128 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: Spark on Kubernetes Integration Test (Resource Limited)
+
+on:
+  # Note:
+  # - '**' if we want to trigger for each pr
+  # - 'master' if we don't want to trigger for each pr
+  # Enable push to do a post validation for each commits in current branch
+  push:
+    branches:
+      - '**'
```

Comment:
If we still think it should be beta, we can change this to `master` to only validate commits after merged to master.